src/hacl_x25519.mli

```diff
@@ -14,19 +14,19 @@
     use this in the context of TLS 1.3.
 *)
 
-val key_length_bytes : int
-(** The length of public and private keys, in bytes. Equal to 32. *)
+(** Key material. In elliptic curve terms, a scalar.
 
-(** A private key. In elliptic curve terms, a scalar.
-
-    To generate a key pair:
-    - generate a random cstruct of length [key_length_bytes].
-    - call [priv_key_of_cstruct] on it. This is the private key.
-    - call [public] on the private key. This returns the corresponding public
-    key.
+    To generate a key pair, use [gen_key].
 *)
 type secret
 
+val gen_key : rng:(int -> Cstruct.t) -> secret * Cstruct.t
+(** Generate a key pair. [rng] is a generic [Cstruct.t] allocator that takes a
+    key length as argument.
```

Comment:
I'd suggest '"key length" in number of bytes' (otherwise, key length is not clearly defined, i.e. could be length in bits)